src/components/buttons/z-chip/index.tsx

```diff
@@ -1,50 +1,80 @@
-import { Component, Prop, h, Element } from "@stencil/core";
-import { ZChipType } from "../../../beans";
+import {
+  Component,
+  Prop,
+  h,
+  Element,
+  Event,
+  EventEmitter,
+} from "@stencil/core";
+import { DeviceEnum, ZChipType } from "../../../beans";
+import { getDevice } from "../../../utils/utils";
 @Component({
   tag: "z-chip",
   styleUrl: "styles.css",
   shadow: true,
 })
 export class ZChip {
-  @Element() hostElement: HTMLElement;
+  @Element() el: HTMLElement;
 
-  @Prop() regulartext?: string;
-  @Prop() boldtext?: number;
-  @Prop({ reflect: true }) type?: ZChipType;
+  @Prop() label?: string;
+  @Prop() icon?: string;
+  @Prop({ reflect: true }) iconPosition?: "left" | "right" = "left";
+  @Prop({ reflect: true }) bold?: boolean = false;
+  @Prop({ reflect: true }) type?: ZChipType = ZChipType.default;
+
+  /** z-chip interaction props */
+  @Prop({ reflect: true }) interaction?: boolean = false;
   @Prop({ reflect: true }) disabled?: boolean = false;
-  @Prop({ reflect: true }) filter?: boolean = false;
 
-  private renderLegacyChip() {
-    return (
-      <div>
-        <span class="boldtext">{this.boldtext}</span> {this.regulartext}
-      </div>
-    );
+  @Event() removeChip: EventEmitter;
+  removeChips() {
+    this.removeChip.emit();
+    this.el.remove();
```

Comment:
In ogni caso se emetti l'evento poi è l'applicazione che lo gestisce che decide cosa fare, non andrebbe bene fare remove